test/L1GovernorMetadataBridge.t.sol

```diff
@@ -4,41 +4,101 @@ pragma solidity ^0.8.0;
+  L2GovernorMetadata l2GovernorMetadata;
+
+  constructor() {
+    setTestnetForkChains(6, 5);
+  }
+
+  function setUpSource() public override {
+    ERC20Votes erc20 = new FakeERC20("GovExample", "GOV");
+    gov = new GovernorMock("Testington Dao", erc20);
+    bridge =
+    new L1GovernorMetadataBridge(address(gov), wormholeCoreFuji, wormholeFujiId, wormholePolygonId);
+  }
+
+  function setUpTarget() public override {
+    l2GovernorMetadata = new L2GovernorMetadata(wormholeCoreMumbai);
+  }
+}
+
+contract Constructor is Test, Constants {
+  function testFork_CorrectlySetAllArgs(address gov) public {
+    L1GovernorMetadataBridge bridge =
+      new L1GovernorMetadataBridge(gov, wormholeCoreFuji, wormholeFujiId, wormholePolygonId);
+    assertEq(address(bridge.GOVERNOR()), gov, "Governor is not set correctly");
+  }
+}
+
+contract Initialize is L1GovernorMetadataBridgeTest {
+  function testFork_CorrectlyInitializeL2GovernorMetadata(address l2GovernorMetadata) public {
+    bridge.initialize(address(l2GovernorMetadata));
+    assertEq(
+      bridge.L2_GOVERNOR_ADDRESS(), l2GovernorMetadata, "L2 governor address is not setup correctly"
+    );
+    assertEq(bridge.INITIALIZED(), true, "Bridge isn't initialized");
+  }
+
+  function testFork_InitlializeL2GovernorMetadataWhenAlreadyInitialized(address l2GovernorMetadata)
+    public
+  {
+    bridge.initialize(address(l2GovernorMetadata));
+
+    vm.expectRevert(L1GovernorMetadataBridge.AlreadyInitialized.selector);
+    bridge.initialize(address(l2GovernorMetadata));
   }
 }
 
 contract Bridge is L1GovernorMetadataBridgeTest {
-  function testFork_CorrectlyBridgeMetadata() public {
-    ERC20Votes erc20 = new FakeERC20("GovExample", "GOV");
-    IGovernor gov = new GovernorMock("Testington Dao", erc20);
-    L1GovernorMetadataBridge bridge = new L1GovernorMetadataBridge(address(gov), wormholeCoreFuji);
-    bridge.initialize(0xBaA85b5C4c74f53c46872acfF2750f512bcBEC43);
+  function testFork_CorrectlyBridgeProposal(uint224 _amount) public {
+    bridge.initialize(address(l2GovernorMetadata));
+    uint256 cost = bridge.quoteDeliveryCost(wormholeFujiId);
+    vm.recordLogs();
 
-    bytes memory mintCalldata = abi.encode(FakeERC20.mint.selector, address(gov), 1 ether);
+    bytes memory proposalCalldata = abi.encode(FakeERC20.mint.selector, address(gov), _amount);
 
     address[] memory targets = new address[](1);
     bytes[] memory calldatas = new bytes[](1);
     uint256[] memory values = new uint256[](1);
 
-    targets[0] = address(erc20);
-    calldatas[0] = mintCalldata;
+    targets[0] = address(fake);
+    calldatas[0] = proposalCalldata;
     values[0] = 0;
 
     // Create proposal
     uint256 proposalId =
       gov.propose(targets, values, calldatas, "Proposal: To inflate governance token");
 
-    uint256 sequence = bridge.bridge(proposalId);
+    bridge.bridge{value: cost}(proposalId);
+    uint256 voteStart = gov.proposalSnapshot(proposalId);
```

Comment:
better names would make this test easier to read. I.e. `layerOneVoteStart` and `layerOneVoteEnd` or similar

Reply:
https://github.com/ScopeLift/l2-flexible-voting/pull/21/commits/9fa5ecdd5cf019e0b669387b340c15e511b452b6